File: frontend/src/types/resume.ts
/** 对齐 ResumeResponse */
export interface ResumeResponse {
  id: number;
  candidateName: string;
  phone: string | null;
  email: string | null;
  rawText: string | null;
  parseStatus: string;
  parsedResume: ParsedResume | null;
  fileUrl: string | null;
  hasEmbedding: boolean;
  createdAt: string;
  updatedAt: string;
}

/** 对齐 ParsedResume */
export interface ParsedResume {
  candidateName: string;
  phone: string | null;
  email: string | null;
  yearsOfExperience: number | null;
  education: string | null;
  currentTitle: string | null;
  skills: string[];
  workExperiences: WorkExperience[];
  projectExperiences: ProjectExperience[];
}

/** 对齐 WorkExperience */
export interface WorkExperience {
  type: 'WORK' | 'INTERNSHIP' | string;
  company: string;
  title: string;
  period: string;
  description: string;
}

/** 对齐 ProjectExperience */
export interface ProjectExperience {
  name: string;
  role: string | null;
  period: string | null;
  description: string | null;
  highlights: string[];
}

/** 对齐 ResumeSearchResult */
export interface ResumeSearchResult {
  id: number;
  candidateName: string;
  phone: string | null;
  email: string | null;
  currentTitle: string | null;
  yearsOfExperience: number | null;
  skills: string[];
  score: number;
  matchedSnippet: string | null;
  embeddingModel: string | null;
}

/** 列表查询参数 */
export interface ResumeQuery {
  page?: number;
  size?: number;
  candidateName?: string;
}

File: interview-core/src/main/java/com/aims/core/resume/Resume.java
package com.aims.core.resume;

import java.time.Instant;

/**
 * 简历领域模型。
 *
 * @param id 简历 ID
 * @param candidateName 候选人姓名
 * @param phone 联系电话
 * @param email 邮箱
 * @param rawText 简历原文
 * @param parsedResume 解析后的结构
 * @param fileUrl 简历文件 URL
 * @param parseStatus 解析状态
 * @param embedding 简历向量
 * @param createdAt 创建时间
 * @param updatedAt 更新时间
 */
public record Resume(
        Long id,
        String candidateName,
        String phone,
        String email,
        String rawText,
        ParsedResume parsedResume,
        String fileUrl,
        ResumeStatus parseStatus,
        String embedding,
        Instant createdAt,
        Instant updatedAt) {}
